sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -113,6 +114,8 @@ class SparkSession private(
 
   private[sql] val sessionUUID: String = UUID.randomUUID.toString
 
+  var jobArtifactState: JobArtifactState = JobArtifactState(sessionUUID, None)
```

Comment:
I think the session state should be immutable. Can we refactor whole things to here? Basically we need to move things `JobArtifactSet` to here. Should probably leverage `InheritableThreadLocal`.

Now `JobArtifactSet` works as a threadlocal. If we want the proper inheritance between sessions, should probably use `InheritableThreadLocal`. 

I think we really need a design here.

Reply:
it sounds good support proper inheritance between sessions, but i no full idea of implement this target, i will mark this pr as draft
can you implement it continue, when complete, we will patch spark official implement to our inner spark version.